### ISSUE TITLE
Move Louvain

### ISSUE THREAD

**Issue by dwaynepryce** [microsoft/topologic#3]
The louvain.py file is one method for creating a global partition structure of the graph. We already plan on including leiden's algorithm for global partitioning, and there could be others.

There's also the ability to partition a graph based off the embeddings, which is currently present in the `topologic.embedding.clustering` package.  A more-unified package like `topologic.partitioning` should probably be created, with louvain.py being moved into it. We should also rename the functions from `best_partition` to `louvain()` and `induce_graph_by_communities` should be specifically made to be not in the louvain.py module. 

In the future I envision this partitioning package to also include ways of taking the output of an embedding clustering and inducing a graph from that as well.

In short, we have some short sighted naming going on, and we should use this opportunity to fix it.
